src/jdk.httpserver/share/classes/sun/net/httpserver/ServerImpl.java

```diff
@@ -521,6 +521,18 @@ private void closeConnection(HttpConnection conn) {
         public void run () {
             /* context will be null for new connections */
             logger.log(Level.TRACE, "exchange started");
+
+            if (dispatcherThread != null && dispatcherThread == Thread.currentThread()) {
```

Comment:
Hello Julia,
Minor thing - With the `dispatcherThread == Thread.currentThread()` check, the `dispatcherThread != null` would be redundant and perhaps can be removed?